contracts/vaults/TreasuryVault/base/RibbonVault.sol

```diff
@@ -0,0 +1,761 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import {SafeMath} from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {
+    SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {
+    ReentrancyGuardUpgradeable
+} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {
+    OwnableUpgradeable
+} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {
+    ERC20Upgradeable
+} from "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+
+import {Vault} from "../../../libraries/Vault.sol";
+import {VaultLifecycle} from "../../../libraries/VaultLifecycle.sol";
+import {
+    VaultLifecycleTreasury
+} from "../../../libraries/VaultLifecycleTreasury.sol";
+import {ShareMath} from "../../../libraries/ShareMath.sol";
+import {IWETH} from "../../../interfaces/IWETH.sol";
+
+contract RibbonVault is
+    ReentrancyGuardUpgradeable,
+    OwnableUpgradeable,
+    ERC20Upgradeable
+{
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+    using ShareMath for Vault.DepositReceipt;
+
+    struct WhitelistInfo {
+        bool isWhitelist;
+        uint256 index;
+    }
```

Comment:
Why do we need to store the index?

Reply:
Im storing the whitelist as an array as well, to iterate for distribution. The index helps when removing address from the array.

Reply:
Since the whitelist array is small, we can [iterate the whitelist and delete it](https://stackoverflow.com/questions/49051856/is-there-a-pop-functionality-for-solidity-arrays)

Reply:
alright